### PR TITLE
Add test on ChatSPN server

### PR DESCRIPTION
**Pull request by Vladyslav-Kuksiuk**
This PR provides tests on `ChatsSpnServer`  through an [issue with subscriptions](https://github.com/spine-examples/ChatSPN/issues/30).

### CODE REVIEW

File: server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+import static io.spine.examples.chatspn.server.given.ChatSpnServerTestEnv.accountCreated;
+import static io.spine.examples.chatspn.server.given.ChatSpnServerTestEnv.createAccount;
+import static io.spine.examples.chatspn.server.given.ChatSpnServerTestEnv.userProfile;
+import static io.spine.util.Exceptions.newIllegalStateException;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@DisplayName("ChatSPN server should")
+final class ChatSpnServerTest {
+
+    private static final int TEST_SERVER_PORT = DEFAULT_CLIENT_SERVICE_PORT;
```
Comment (Artem-Semenov-dev) [inline server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java:62]: I think it is redundant, you can just use `DEFAULT_CLIENT_SERVICE_PORT`.

File: server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+              .postAndForget();
+        var expectedEvent = accountCreated(secondCommand);
+
+        assertExpectedFields(futureEvent.get(2, SECONDS), expectedEvent);
+    }
+
+    /**
+     * Creates a new client with access to the server.
+     */
+    Client createClient() {
```
Comment (Artem-Semenov-dev) [inline server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java:202]: It can be private.

**Review by Artem-Semenov-dev — COMMENTED** [review]
@Vladyslav-Kuksiuk almost LGTM. Please see my comments.

File: server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+                channel.awaitTermination(1, SECONDS);
+            } catch (InterruptedException e) {
+                throw newIllegalStateException(e, "Channel can't be terminated");
+            }
+        });
+        server.shutdown();
+    }
+
+    @Test
+    @DisplayName("handle command and request for projection status")
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java:95]: "Call a method and then call another method, receiving some Proto message is response"?

File: server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                .asGuest()
+                .select(UserProfile.class)
+                .run();
+        var expectedUserProfile = userProfile(command);
+
+        assertExpectedFields(userProfileResponse.get(0), expectedUserProfile);
+    }
+
+    @Test
+    @DisplayName("handle subscription on entity")
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java:112]: Continuing my previous comment.

Let's speak in domain terms. I would also refrain from saying "handle" everywhere.

**Review by armiol — CHANGES_REQUESTED** [review]
@Vladyslav-Kuksiuk please see my comments.

File: server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+            } catch (InterruptedException e) {
+                throw newIllegalStateException(e, "Channel can't be terminated");
+            }
+        });
+        server.shutdown();
+    }
+
+    @Test
+    @DisplayName("accept the `CreateAccount` command and return `UserProfile` projection state" +
+            "in response to `QueryRequest` by the `UserProfile` class")
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java:96]: There is a space missing either here or in the previous line.

File: server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+              .postAndForget();
+        var expectedEvent = accountCreated(command);
+
+        assertExpectedFields(futureEvent.get(2, SECONDS), expectedEvent);
+    }
+
+    @Test
+    @DisplayName("subscribe to the `AccountCreated` event by ID " +
+            "and call 'observing consumer' when the `AccountCreated` event with provided 'id' field " +
+            "will be emitted in response to the `CreateAccount` command")
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java:183]: "is emitted".

It's here and in other similar cases.

Please have a read on conditionals in English.

**Pull request merged**